qutebrowser/config/configdata.py

```diff
@@ -402,6 +402,10 @@ def data(readonly=False):
              "Whether to shrink the completion to be smaller than the "
              "configured size if there are no scrollbars."),
 
+            ('match-type',
+             SettingValue(typ.String(none_ok=True), 'contain'),
```

Comment:
Ah, now I see why you checked for invalid values earlier - this should be its own config type which validates and documents the possible values.

Add something like this at the bottom of `config/configtypes.py`:

```python
class MatchType(BaseType):

    """How to match values in the completion."""

    valid_values = ValidValues(('contain', "Match items which contain the "
                                           "entered text."),
                               ('fuzzy', "Match items with a fuzzy matching "
                                         "logic."),
                               ('start', "Only match items starting with the "
                                         "entered text."),
```

And then use `typ.MatchType()`.